tests/test_manager.py

```diff
@@ -124,13 +125,65 @@ def test_ConfigIniEnv(datadir):
     ini_filename = os.path.join(datadir, 'config_test.ini')
     cie = ConfigIniEnv([ini_filename])
     assert cie.get('foo') == 'bar'
-    assert cie.get('foo', namespace=['namespacebaz']) == 'bat'
+    assert cie.get('foo', namespace='namespacebaz') == 'bat'
     # FIXME: Add multi-tier namespace
 
     cie = ConfigIniEnv(['/a/b/c/bogus/filename'])
     assert cie.get('foo') == NO_VALUE
 
 
+def test_ConfigEnvFileEnv(datadir):
+    env_filename = os.path.join(datadir, '.env')
+    cefe = ConfigEnvFileEnv(['/does/not/exist/.env', env_filename])
+    assert cefe.get('not_a', namespace='youre') == 'golfer'
+    assert cefe.get('loglevel') == 'walter'
+    assert cefe.get('missing') is NO_VALUE
+    assert cefe.data == {
+        'LOGLEVEL': 'walter',
+        'DEBUG': 'True',
+        'YOURE_NOT_A': 'golfer',
+        'DATABASE_URL': 'sqlite:///kahlua.db',
+    }
+
+    cefe = ConfigEnvFileEnv('/does/not/exist/.env')
+    assert cefe.get('loglevel') is NO_VALUE
+
+
+def test_parse_env_file():
+    assert parse_env_file(['PLAN9=outerspace']) == {'PLAN9': 'outerspace'}
+    with pytest.raises(ConfigurationError):
+        parse_env_file(['3AMIGOS=infamous'])
+    with pytest.raises(ConfigurationError):
+        parse_env_file(['INVALID-CHAR=value'])
+    with pytest.raises(ConfigurationError):
+        parse_env_file(['MISSING-equals'])
+
+
+def test_get_key_from_envs():
+    assert get_key_from_envs({'K': 'v'}, 'k') == 'v'
+    assert get_key_from_envs([{'K': 'v'},
+                              {'L': 'w'}], 'l') == 'w'
+    assert get_key_from_envs({'T_T': 'sad'},
+                             't', namespace='t') == 'sad'
+    assert get_key_from_envs({'O_A_O': 'strange'},
+                             'o', namespace=['o', 'a']) == 'strange'
+    assert get_key_from_envs({'K': 'v'}, 'q') is NO_VALUE
+    # first match wins
+    assert get_key_from_envs([
+        {'K': 'v'},
+        {'L': 'w'},
+        {'K': 'z'},
+    ], 'k') == 'v'
+    # works with reversed iterator
+    assert get_key_from_envs(reversed([
```

Comment:
I've thought a lot about what to do with style problems here where assert is a statement but you've got a really long line. Generally, I want the end result to be easily parsed into "this == that". My current style is to do something like this:

```
assert (
    get_keys_from_envs(reversed([{'L': 'v'}, {'L': 'w'},]), '1') ==
    'w'
)
```

That form works pretty well for long lines because it puts the expression far to the left plus it breaks up the "this" and "that" into separate lines. If the "this" expression got even longer, then you get to just continue using normal indenting rules.

To clarify, I'm not advocating we change this. It's just something I think about periodically and figured I'd share.